src/include/storage/sql_table.h

```diff
@@ -62,8 +64,14 @@ class SqlTable {
    * @param out_buffer output buffer. The object should already contain projection list information. @see ProjectedRow.
    * @return true if tuple is visible to this txn and ProjectedRow has been populated, false otherwise
    */
-  bool Select(const common::ManagedPointer<transaction::TransactionContext> txn, const TupleSlot slot,
+  bool Select(const common::ManagedPointer <transaction::TransactionContext> txn,
+              layout_version_t layout_version,
+              const TupleSlot slot,
               ProjectedRow *const out_buffer) const {
+    // TODO(Schema-Change): Modify this
+    //  a expected tuple can be in the dataTable of any version, as they do not overlap.
+    //  need to go through all datatable prior or equal to the current schema and do Select for each one
+    //  Try to do the traversal N2O
```

Comment:
strictly speaking, we don't need to look beyond the dataTables of versions larger than 'layout_version'

Reply:
dui